quinn/src/connection.rs

```diff
@@ -364,6 +366,30 @@ impl Connection {
         }
     }
 
+    /// Accept the next incoming uni-directional stream
```

Comment:
> Is this intended to replace the IncomingUniStreams/IncomingBiStreams stuff?

Right. My thinking is that this makes the API a bit simpler, primarily since `NewConnection` can be dispensed with entirely. On the other hand, it makes it much more difficult to use off-the-shelf stream combinators like [`buffer_unordered`](https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.buffer_unordered). For the most part I think combinators have rightly fallen out of fashion compared to imperative-style `async` blocks, but the constructs aren't always exactly equivalent; e.g. emulating `buffer_unordered` might require combining `tokio::spawn` with an async semaphore, which would not be as graceful about memory reuse, but saves a layer of indirection on the wake-up path.

> Should we deprecate that API?

In principle this would be preferred, but deprecating `NewConnection` gracefully seems difficult, since it's central to accepting/creating new connections; e.g. `Endpoint::connect` itself would need to be deprecated, which consumes an awfully appealing part of the namespace.